### PR TITLE
FIX-#6070: Simplify implementation of `shift`

### LINKED ISSUE #6070: REFACTOR: Fix `df.shift`
- https://github.com/ponder-org/modin-public/commit/73bd234fc89f89501928afc3267e6d3f83cb5977
- https://github.com/ponder-org/modin-public/commit/3d9516f804224c30cb8bb20f1559eceb051c3542

These should be upstreamed, despite the commit message.

### PR DESCRIPTION
**Pull request by RehanSD**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #6070 ? 
- [ ] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/base.py [hunk 1]
```diff
@@ -2833,71 +2833,16 @@ def shift(
...
-            else:
-                fill_columns = self.columns
-
-        filled_df = (
-            self.__constructor__(index=fill_index, columns=fill_columns)
-            if isinstance(self, DataFrame)
-            else self.__constructor__(index=fill_index, name=self.name)
+        return (
+            self._create_or_update_from_compiler(
+                new_query_compiler=self._query_compiler.shift(
```
Comment (anmyachev) [inline modin/pandas/base.py:2838]: @RehanSD looks like `_query_compiler` doesn't have `shift` method.

**Comment by mvashishtha** [comment]
@RehanSD I think you should implement `shift` in the base and pandas query compilers. The base implementation should translate the existing API layer implementation.

**Review by devin-petersohn — COMMENTED** [review]
I added `shift` to the `BaseQueryCompiler` as a `Default` method. Previously we were pretending to do distributed things here, but this line actually means that we are not really doing anything distributed: https://github.com/modin-project/modin/blob/492934c078a01ecf18e70c25fa7b1e2d8b8271a0/modin/pandas/base.py#L2860-L2864

**Comment by RehanSD** [comment]
> I added `shift` to the `BaseQueryCompiler` as a `Default` method. Previously we were pretending to do distributed things here, but this line actually means that we are not really doing anything distributed:
> 
> https://github.com/modin-project/modin/blob/492934c078a01ecf18e70c25fa7b1e2d8b8271a0/modin/pandas/base.py#L2860-L2864

I'm not sure that `shift` is defaulting. Although we do call the constructor with a new index + new column, we also do operations on the new (empty) dataframe. Although the first fillna may default since the dataframe is empty, the remaining operations (e.g. the drop) will not be defaulted right?

**Comment by devin-petersohn** [comment]
>I'm not sure that shift is defaulting. Although we do call the constructor with a new index + new column, we also do operations on the new (empty) dataframe. Although the first fillna may default since the dataframe is empty, the remaining operations (e.g. the drop) will not be defaulted right?

The dataframe will not be empty, it will be full of null values, so we are in effect defaulting to pandas on the constructor and then pretending to be distributed. I think with a query compiler method this will encourage a true distributed implementation.

**Review by mvashishtha — APPROVED** [review]

**Pull request merged**